### PR TITLE
React 18

### PR DESCRIPTION
**Pull request by viet-nv**
Upgrade:
- React 18: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html
- react-scripts v4 (unfortunately, react-scripts 5 is not working)
- Typescript v4
- Styled component v5

### CODE REVIEW

File: src/components/AddToMetamask/index.tsx [hunk 1]
```diff
@@ -1,10 +1,9 @@
-import React from 'react'
-import styled from 'styled-components'
 import { ChainId, Token } from '@kyberswap/ks-sdk-core'
-
 import MetaMaskLogo from 'assets/images/metamask.png'
 import { ButtonEmpty } from 'components/Button'
 import { RowFixed } from 'components/Row'
+import React from 'react'
+import styled from 'styled-components'
```
Comment (namgold) [inline src/components/AddToMetamask/index.tsx:6]: lol need to check this

Comment (namgold) [reply src/components/AddToMetamask/index.tsx:6]: it sorted from nice order to alphabet

Comment (namgold) [reply src/components/AddToMetamask/index.tsx:6]: should remove it

Comment (viet-nv) [reply src/components/AddToMetamask/index.tsx:6]: we should have a common rule for sorting import 
And this rule is good

Comment (namgold) [reply src/components/AddToMetamask/index.tsx:6]: Let's seek for other better rule

File: src/components/MenuFlyout/index.tsx [hunk 2]
```diff
@@ -80,23 +82,19 @@ const MenuFlyout = (props: {
   hasArrow?: boolean
 }) => {
   useOnClickOutside(props.node, props.isOpen && !isMobile ? props.toggle : undefined)
-  const BrowserStyle = useMemo(
```
Comment (hungdoansy) [inline src/components/MenuFlyout/index.tsx:83]: I think we should keep useMemo. This PR shouldn't update logics

Comment (viet-nv) [reply src/components/MenuFlyout/index.tsx:83]: this violate rule of hook

Comment (hungdoansy) [reply src/components/MenuFlyout/index.tsx:83]: This is probably because the SC is used dynamically (inside another component)

**Review by namgold — APPROVED** [review]

**Pull request merged**